src/index.js

```diff
@@ -56,37 +64,46 @@ const requestFactory = (meth, notify = null) => {
       finalHeaders['Authorization'] = 'Bearer ' + token
     }
 
+    if (!('rtype' in options)){
+      options['rtype'] = 'json'
+    } else if (['json', 'file', 'string'].indexOf(options['rtype']) < 0){
```

Comment:
Would be best to declare the array of return types at the top and reference them from there.

Reply:
Yeah sure. Why do you think that though? My thinking was there is no need to keep this array around after the if statement so using it like this would make sure it was destroyed after the if block scope. I just though that seemed a bit better then having it hand around for the scope of the function?

Reply:
It's better from a maintenance and readability perspective. Declaring it at the top makes it obvious what the allowed return types are instead of having to search through the code.

As long as you use `const` and name the array something clear and specific, there's no real issues with it persisting.

Referencing the array through the function keeps the code DRY and makes it more maintainable. For example, if someone wanted to add another return type it would be much more straightforward.

Reply:
Yeah ok sure